lib/swoc/CMakeLists.txt

```diff
@@ -118,3 +118,6 @@ set_target_properties(
 )
 
 install(TARGETS libswoc PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/swoc)
+
+add_library(libswoc::libswoc ALIAS libswoc-static)
+add_library(libswoc::libswoc ALIAS libswoc)
```

Comment:
Whoa there. This looks problematic. 😄

Reply:
That was the easy fix...  I am happy to have a better fix for the cmake build file for libswoc

Reply:
Maybe you can provide some context on why it was done this way. If we decide to keep this, we should definitely comment it, or this will be a "whoa there" moment for everyone who lays eyes on this in the future.

If we want to automatically pick up either the static or dynamic libswoc, depending on which one was found, then we do want one target that could represent either. How come this even works? Some projects have both a static and dynamic target and install both of them - what does CMake do if both targets exist? Does `libswoc::libswoc` end up aliasing the static one, or the dynamic one? Or does CMake error out?

Reply:
I think he should use `libswoc::libswoc-static` for the static variant.

Reply:
@cmcfarlen I believe was saying that we want to say something different to tell CMake what to use for the library